src/twisted/conch/test/test_userauth.py

```diff
@@ -207,13 +208,28 @@ class PrivateKeyChecker:
     credentialInterfaces = (ISSHPrivateKey,)
 
     def requestAvatarId(self, creds):
-        if creds.blob == keys.Key.fromString(keydata.publicRSA_openssh).blob():
+        # Accept both RSA and Security Key public keys for testing
+        validBlobs = [
+            keys.Key.fromString(keydata.publicRSA_openssh).blob(),
+        ]
+
+        if creds.blob in validBlobs:
             if creds.signature is not None:
                 obj = keys.Key.fromString(creds.blob)
                 if obj.verify(creds.signature, creds.sigData):
                     return creds.username
             else:
                 raise ValidPublicKey()
+        # blobs are generated randomly using SKDummy
+        elif creds.algName.startswith(b"sk-"):
+            if creds.signature is not None:
+                obj = keys.Key.fromString(creds.blob)
+                if obj.verify(creds.signature, creds.sigData):
+                    return creds.username
+                else:
+                    return UnauthorizedLogin()
```

Comment:
Very minor comment. Do we need the else here ?
I know that existing code is written with else... but since we use `return` I don't think that we need it.

```suggestion

                return UnauthorizedLogin()
```